lib/origin.ts

```diff
@@ -0,0 +1,25 @@
+import { headers } from "next/headers";
+
+/**
+ * Extract the origin from either `AUTH_URL` or `NEXTAUTH_URL` environment variables,
+ * or the request's headers option.
+ */
+export function getOrigin(): string {
+  const envUrl = process.env.AUTH_URL ?? process.env.NEXTAUTH_URL;
+  const h = headers();
+
+  let url: URL;
+  if (envUrl) {
+    url = new URL(envUrl);
+  } else {
+    const detectedHost = h.get("x-forwarded-host") ?? h.get("host");
```

Comment:
Do you think this could be exploited?

I'm looking into this a bit on my end.

Reply:
There is a potential for it to be exploited which is why it's only used on our PR Review and local environments.   In our live environments the NEXTAUTH_URL or AUTH_URL will always be available.